Importing most important python libraries:

In [196]:
import pandas as pd
import numpy as np

There are 8 different csv files* regarding each year. 
<br>To make full analysis throughout 2015-2022, I am going to join each file into main dataframe using 'for in' loop. 

In [197]:
files = ['2015', '2016', '2017', '2018', '2019', '2020', '2021','2022']
          
main_df = pd.DataFrame(pd.read_csv(r"C:/Users/Agata/Documents/GitHub/Happiness-Report/source_files/2015.csv"))
main_df['year'] = '2015'
for i in range(1, len(files)):
    data = pd.read_csv(r"C:/Users/Agata/Documents/GitHub/Happiness-Report/source_files/"+files[i]+r".csv")
    df = pd.DataFrame(data)
    df['year'] = files[i]
    main_df = pd.concat([main_df,df],axis=0)
    main_df = main_df.reset_index(drop=True)
main_df.head()


Country          Region  Happiness Rank  Happiness Score  \
0  Switzerland  Western Europe             1.0            7.587   
1      Iceland  Western Europe             2.0            7.561   
2      Denmark  Western Europe             3.0            7.527   
3       Norway  Western Europe             4.0            7.522   
4       Canada   North America             5.0            7.427   

   Standard Error  Economy (GDP per Capita)   Family  \
0         0.03411                   1.39651  1.34951   
1         0.04884                   1.30232  1.40223   
2         0.03328                   1.32548  1.36058   
3         0.03880                   1.45900  1.33095   
4         0.03553                   1.32629  1.32261   

   Health (Life Expectancy)  Freedom  Trust (Government Corruption)  ...  \
0                   0.94143  0.66557                        0.41978  ...   
1                   0.94784  0.62877                        0.14145  ...   
2                   0.87464  0.64938                        0.48357  ...   
3                   0.88521  0.66973                        0.36503  ...   
4                   0.90563  0.63297                        0.32957  ...   

   Explained by: Freedom to make life choices  Explained by: Generosity  \
0                                         NaN                       NaN   
1                                         NaN                       NaN   
2                                         NaN                       NaN   
3                                         NaN                       NaN   
4                                         NaN                       NaN   

  Explained by: Perceptions of corruption  Dystopia + residual  RANK  \
0                                     NaN                  NaN   NaN   
1                                     NaN                  NaN   NaN   
2                                     NaN                  NaN   NaN   
3                                     NaN                  NaN   NaN   
4                                     NaN                  NaN   NaN   

   Happiness score  Whisker-high  Whisker-low  Dystopia (1.83) + residual  \
0              NaN           NaN          NaN                         NaN   
1              NaN           NaN          NaN                         NaN   
2              NaN           NaN          NaN                         NaN   
3              NaN           NaN          NaN                         NaN   
4              NaN           NaN          NaN                         NaN   

   Explained by: GDP per capita  
0                           NaN  
1                           NaN  
2                           NaN  
3                           NaN  
4                           NaN  

[5 rows x 57 columns]

In [198]:
main_df.columns

Index(['Country', 'Region', 'Happiness Rank', 'Happiness Score',
       'Standard Error', 'Economy (GDP per Capita)', 'Family',
       'Health (Life Expectancy)', 'Freedom', 'Trust (Government Corruption)',
       'Generosity', 'Dystopia Residual', 'year', 'Lower Confidence Interval',
       'Upper Confidence Interval', 'Happiness.Rank', 'Happiness.Score',
       'Whisker.high', 'Whisker.low', 'Economy..GDP.per.Capita.',
       'Health..Life.Expectancy.', 'Trust..Government.Corruption.',
       'Dystopia.Residual', 'Overall rank', 'Country or region', 'Score',
       'GDP per capita', 'Social support', 'Healthy life expectancy',
       'Freedom to make life choices', 'Perceptions of corruption',
       'Country name', 'Regional indicator', 'Ladder score',
       'Standard error of ladder score', 'upperwhisker', 'lowerwhisker',
       'Logged GDP per capita', 'Social support_', 'Healthy life expectancy_',
       'Freedom to make life choices_', 'Generosity_',
       'Perceptions of corr

Let's see what the dataframe looks like:

In [199]:
main_df.info()
main_df.shape

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1231 entries, 0 to 1230
Data columns (total 57 columns):
 #   Column                                      Non-Null Count  Dtype  
---  ------                                      --------------  -----  
 0   Country                                     617 non-null    object 
 1   Region                                      315 non-null    object 
 2   Happiness Rank                              315 non-null    float64
 3   Happiness Score                             315 non-null    float64
 4   Standard Error                              158 non-null    float64
 5   Economy (GDP per Capita)                    315 non-null    float64
 6   Family                                      470 non-null    float64
 7   Health (Life Expectancy)                    315 non-null    float64
 8   Freedom                                     470 non-null    float64
 9   Trust (Government Corruption)               315 non-null    float64
 10  Generosity  

(1231, 57)

The df has 1231 rows and 56 columns. Let's see how many columns are unique:

In [200]:
len(main_df.columns.unique())

57

All columns are unique. However, the number of columns is high so we will focus on null values next:

In [201]:
main_df.isnull().sum(axis=0)

Country                                        614
Region                                         916
Happiness Rank                                 916
Happiness Score                                916
Standard Error                                1073
Economy (GDP per Capita)                       916
Family                                         761
Health (Life Expectancy)                       916
Freedom                                        761
Trust (Government Corruption)                  916
Generosity                                     449
Dystopia Residual                              916
year                                             0
Lower Confidence Interval                     1074
Upper Confidence Interval                     1074
Happiness.Rank                                1076
Happiness.Score                               1076
Whisker.high                                  1076
Whisker.low                                   1076
Economy..GDP.per.Capita.       

We can see that there are more null values in almost each column than non-null values. That means the columns in each file could be named differently. 

In [202]:
main_df.columns.sort_values()


Index(['Country', 'Country name', 'Country or region',
       'Dystopia (1.83) + residual', 'Dystopia + residual',
       'Dystopia Residual', 'Dystopia.Residual', 'Economy (GDP per Capita)',
       'Economy..GDP.per.Capita.',
       'Explained by: Freedom to make life choices',
       'Explained by: GDP per capita', 'Explained by: Generosity',
       'Explained by: Healthy life expectancy',
       'Explained by: Log GDP per capita',
       'Explained by: Perceptions of corruption',
       'Explained by: Social support', 'Family', 'Freedom',
       'Freedom to make life choices', 'Freedom to make life choices_',
       'GDP per capita', 'Generosity', 'Generosity_', 'Happiness Rank',
       'Happiness Score', 'Happiness score', 'Happiness.Rank',
       'Happiness.Score', 'Health (Life Expectancy)',
       'Health..Life.Expectancy.', 'Healthy life expectancy',
       'Healthy life expectancy_', 'Ladder score', 'Ladder score in Dystopia',
       'Logged GDP per capita', 'Lower Confidence 

There is significant redundancy in the columns.
<br>For example, information about countries is split into three columns:
* Country
* Country or region
* Country name
<br> The files have between 9-20 columns. We could equalize the names manually (directly in csv files) which would take long time or create a dictionary with desired names and swap them as below:

In [203]:
correct_names = {'Economy (GDP per Capita)' : 'GDP per Capita',
'Family' : 'Social Support',
'Explained by: Generosity' : "Generosity",
'Health (Life Expectancy)' : 'Healthy Life Expectancy',
'Freedom' : 'Freedom to Make Life Choices',
'Trust (Government Corruption)' : 'Corruption Perception',
'Happiness.Rank' : 'Happiness Rank',
'Happiness.Score' : 'Happiness Score',
'Economy..GDP.per.Capita.' : 'GDP per Capita',
'Health..Life.Expectancy.' : 'Healthy Life Expectancy',
'Trust..Government.Corruption.' : 'Corruption Perception',
'Dystopia.Residual' : 'Dystopia Residual',
'Overall rank' : 'Happiness Rank',
'Country or region' : 'Country',
'Score' : 'Happiness Score',
'GDP per capita' : 'GDP per Capita',
'Country name' : 'Country',
'Regional indicator' : 'Region',
'RANK' : 'Happiness Rank',
'Happiness score' : 'Happiness Score',
'Dystopia (1.83) + residual' : 'Dystopia Residual',
'Ladder score' : 'Happiness Score',
'Dystopia + residual' : 'Dystopia Residual',
'Social support' : 'Social Support',
'Explained by: Social support' : 'Social Support',
'Healthy life expectancy' : 'Healthy Life Expectancy',
'Explained by: Healthy life expectancy' : 'Healthy Life Expectancy',
'Freedom to make life choices' : 'Freedom to Make Life Choices',
'Explained by: Freedom to make life choices' : 'Freedom to Make Life Choices',
'Perceptions of corruption' : 'Corruption Perception',
'Explained by: Perceptions of corruption' : 'Corruption Perception',
'Explained by: Log GDP per capita' : 'GDP per Capita',
'Explained by: GDP per capita' : 'GDP per Capita'
}

In [204]:
main_df = main_df.rename(columns=correct_names)
main_df.columns.sort_values()
main_df.tail()


Country Region  Happiness Rank  Happiness Score  Standard Error  \
1226      Rwanda*    NaN             NaN              NaN             NaN   
1227     Zimbabwe    NaN             NaN              NaN             NaN   
1228      Lebanon    NaN             NaN              NaN             NaN   
1229  Afghanistan    NaN             NaN              NaN             NaN   
1230           xx    NaN             NaN              NaN             NaN   

      GDP per Capita  Social Support  Healthy Life Expectancy  \
1226             NaN             NaN                      NaN   
1227             NaN             NaN                      NaN   
1228             NaN             NaN                      NaN   
1229             NaN             NaN                      NaN   
1230             NaN             NaN                      NaN   

      Freedom to Make Life Choices Corruption Perception  ...  \
1226                           NaN                   NaN  ...   
1227                           NaN                   NaN  ...   
1228                           NaN                   NaN  ...   
1229                           NaN                   NaN  ...   
1230                           NaN                   NaN  ...   

      Freedom to Make Life Choices  Generosity Corruption Perception  \
1226                         0.621       0.187                 0.544   
1227                         0.329       0.106                 0.105   
1228                         0.103       0.082                 0.034   
1229                         0.000       0.089                 0.005   
1230                           NaN         NaN                   NaN   

      Dystopia Residual  Happiness Rank  Happiness Score  Whisker-high  \
1226                NaN           143.0            3.268         3.462   
1227                NaN           144.0            2.995         3.110   
1228                NaN           145.0            2.955         3.049   
1229                NaN           146.0            2.404         2.469   
1230                NaN           147.0              NaN           NaN   

      Whisker-low  Dystopia Residual  GDP per Capita  
1226        3.074              0.001           0.001  
1227        2.880              0.001           0.001  
1228        2.862              0.000           1.392  
1229        2.339              1.263           0.001  
1230          NaN                NaN             NaN  

[5 rows x 57 columns]

In [205]:
files = ['2015', '2016', '2017', '2018', '2019', '2020','2021','2022']
          
main_df = pd.DataFrame(pd.read_csv(r"C:/Users/Agata/Documents/GitHub/Happiness-Report/source_files/2015.csv"))
main_df = main_df.rename(columns=correct_names)
main_df['Year'] = '2015'

for i in range(1, len(files)):
    data = pd.read_csv(r"C:/Users/Agata/Documents/GitHub/Happiness-Report/source_files/"+files[i]+r".csv")
    df = pd.DataFrame(data)
    df = df.rename(columns=correct_names)
    df['Year'] = files[i]
    main_df = pd.concat([main_df,df],axis=0)
    main_df = main_df.reset_index(drop=True)
    df.to_csv(r"C:/Users/Agata/Documents/"+files[i]+r".csv", index=True)

main_df['Year'] = main_df['Year'].astype(int)
main_df.tail()

#df = pd.DataFrame(pd.read_csv(r"C:/Users/Agata/Documents/GitHub/Happiness-Report/source_files/2016.csv"))
#df = df.rename(columns=correct_names)
#df['Year'] = '2016'
#main_df = pd.concat([main_df,df],axis=0)
#main_df = main_df.reset_index(drop=True)
#main_df =  main_df.set_index(['Year','Country'])
#main_df.to_csv(r"C:/Users/Agata/Documents/"+files[i]+r".csv", index=True)

Country Region  Happiness Rank  Happiness Score  Standard Error  \
1226      Rwanda*    NaN           143.0            3.268             NaN   
1227     Zimbabwe    NaN           144.0            2.995             NaN   
1228      Lebanon    NaN           145.0            2.955             NaN   
1229  Afghanistan    NaN           146.0            2.404             NaN   
1230           xx    NaN           147.0              NaN             NaN   

      GDP per Capita  Social Support  Healthy Life Expectancy  \
1226           0.001           0.000                    0.462   
1227           0.001           0.001                    0.270   
1228           1.392           0.000                    0.631   
1229           0.001           0.000                    0.289   
1230             NaN             NaN                      NaN   

      Freedom to Make Life Choices  Corruption Perception  ...  lowerwhisker  \
1226                         0.621                  0.544  ...           NaN   
1227                         0.329                  0.105  ...           NaN   
1228                         0.103                  0.034  ...           NaN   
1229                         0.000                  0.005  ...           NaN   
1230                           NaN                    NaN  ...           NaN   

      Logged GDP per capita  Social support_  Healthy life expectancy_  \
1226                    NaN              NaN                       NaN   
1227                    NaN              NaN                       NaN   
1228                    NaN              NaN                       NaN   
1229                    NaN              NaN                       NaN   
1230                    NaN              NaN                       NaN   

      Freedom to make life choices_  Generosity_  Perceptions of corruption_  \
1226                            NaN          NaN                         NaN   
1227                            NaN          NaN                         NaN   
1228                            NaN          NaN                         NaN   
1229                            NaN          NaN                         NaN   
1230                            NaN          NaN                         NaN   

      Ladder score in Dystopia  Whisker-high  Whisker-low  
1226                       NaN         3.462        3.074  
1227                       NaN         3.110        2.880  
1228                       NaN         3.049        2.862  
1229                       NaN         2.469        2.339  
1230                       NaN           NaN          NaN  

[5 rows x 29 columns]

In [206]:
main_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1231 entries, 0 to 1230
Data columns (total 29 columns):
 #   Column                          Non-Null Count  Dtype  
---  ------                          --------------  -----  
 0   Country                         1231 non-null   object 
 1   Region                          617 non-null    object 
 2   Happiness Rank                  929 non-null    float64
 3   Happiness Score                 1230 non-null   float64
 4   Standard Error                  158 non-null    float64
 5   GDP per Capita                  1230 non-null   float64
 6   Social Support                  1230 non-null   float64
 7   Healthy Life Expectancy         1230 non-null   float64
 8   Freedom to Make Life Choices    1230 non-null   float64
 9   Corruption Perception           1229 non-null   float64
 10  Generosity                      1230 non-null   float64
 11  Dystopia Residual               918 non-null    float64
 12  Year                            12

In [213]:
df = main_df.loc[:,:'Year']
df = df.drop(columns='Standard Error')
df = df.drop(axis=0, index=1230)
df['Country'] = df['Country'].str.replace('*','')
df.loc[df.Year.isin([2020,2021])]

C:\Users\Agata\AppData\Local\Temp\ipykernel_11328\819183918.py:4: FutureWarning: The default value of regex will change from True to False in a future version. In addition, single character regular expressions will *not* be treated as literal strings when regex=True.
  df['Country'] = df['Country'].str.replace('*','')


Country              Region  Happiness Rank  Happiness Score  \
782       Finland      Western Europe             NaN           7.8087   
783       Denmark      Western Europe             NaN           7.6456   
784   Switzerland      Western Europe             NaN           7.5599   
785       Iceland      Western Europe             NaN           7.5045   
786        Norway      Western Europe             NaN           7.4880   
...           ...                 ...             ...              ...   
1079      Lesotho  Sub-Saharan Africa             NaN           3.5120   
1080     Botswana  Sub-Saharan Africa             NaN           3.4670   
1081       Rwanda  Sub-Saharan Africa             NaN           3.4150   
1082     Zimbabwe  Sub-Saharan Africa             NaN           3.1450   
1083  Afghanistan          South Asia             NaN           2.5230   

      GDP per Capita  Social Support  Healthy Life Expectancy  \
782         1.285190        1.499526                 0.961271   
783         1.326949        1.503449                 0.979333   
784         1.390774        1.472403                 1.040533   
785         1.326502        1.547567                 1.000843   
786         1.424207        1.495173                 1.008072   
...              ...             ...                      ...   
1079        0.451000        0.731000                 0.007000   
1080        1.099000        0.724000                 0.340000   
1081        0.364000        0.202000                 0.407000   
1082        0.457000        0.649000                 0.243000   
1083        0.370000        0.000000                 0.126000   

      Freedom to Make Life Choices  Corruption Perception  Generosity  \
782                       0.662317               0.477857    0.159670   
783                       0.665040               0.495260    0.242793   
784                       0.628954               0.407946    0.269056   
785                       0.661981               0.144541    0.362330   
786                       0.670201               0.434101    0.287985   
...                            ...                    ...         ...   
1079                      0.405000               0.015000    0.103000   
1080                      0.539000               0.088000    0.027000   
1081                      0.627000               0.493000    0.227000   
1082                      0.359000               0.075000    0.157000   
1083                      0.000000               0.010000    0.122000   

      Dystopia Residual  Year  
782            2.762835  2020  
783            2.432741  2020  
784            2.350267  2020  
785            2.460688  2020  
786            2.168266  2020  
...                 ...   ...  
1079           1.800000  2021  
1080           0.648000  2021  
1081           1.095000  2021  
1082           1.205000  2021  
1083           1.895000  2021  

[302 rows x 12 columns]

In [214]:
rank = df[df.Year.isin([2020,2021])]
rank['Happiness Rank'] = rank.groupby('Year')['Happiness Score'].rank('dense',ascending=False)
#rank.to_csv(r'C:/Users/Agata/Desktop/test_rank.csv', index=True)
df.loc[782:1083,'Happiness Rank'] = rank['Happiness Rank']
df[df.Year.isin([2020,2021])]

C:\Users\Agata\AppData\Local\Temp\ipykernel_11328\1998301519.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  rank['Happiness Rank'] = rank.groupby('Year')['Happiness Score'].rank('dense',ascending=False)


Country              Region  Happiness Rank  Happiness Score  \
782       Finland      Western Europe             1.0           7.8087   
783       Denmark      Western Europe             2.0           7.6456   
784   Switzerland      Western Europe             3.0           7.5599   
785       Iceland      Western Europe             4.0           7.5045   
786        Norway      Western Europe             5.0           7.4880   
...           ...                 ...             ...              ...   
1079      Lesotho  Sub-Saharan Africa           143.0           3.5120   
1080     Botswana  Sub-Saharan Africa           144.0           3.4670   
1081       Rwanda  Sub-Saharan Africa           145.0           3.4150   
1082     Zimbabwe  Sub-Saharan Africa           146.0           3.1450   
1083  Afghanistan          South Asia           147.0           2.5230   

      GDP per Capita  Social Support  Healthy Life Expectancy  \
782         1.285190        1.499526                 0.961271   
783         1.326949        1.503449                 0.979333   
784         1.390774        1.472403                 1.040533   
785         1.326502        1.547567                 1.000843   
786         1.424207        1.495173                 1.008072   
...              ...             ...                      ...   
1079        0.451000        0.731000                 0.007000   
1080        1.099000        0.724000                 0.340000   
1081        0.364000        0.202000                 0.407000   
1082        0.457000        0.649000                 0.243000   
1083        0.370000        0.000000                 0.126000   

      Freedom to Make Life Choices  Corruption Perception  Generosity  \
782                       0.662317               0.477857    0.159670   
783                       0.665040               0.495260    0.242793   
784                       0.628954               0.407946    0.269056   
785                       0.661981               0.144541    0.362330   
786                       0.670201               0.434101    0.287985   
...                            ...                    ...         ...   
1079                      0.405000               0.015000    0.103000   
1080                      0.539000               0.088000    0.027000   
1081                      0.627000               0.493000    0.227000   
1082                      0.359000               0.075000    0.157000   
1083                      0.000000               0.010000    0.122000   

      Dystopia Residual  Year  
782            2.762835  2020  
783            2.432741  2020  
784            2.350267  2020  
785            2.460688  2020  
786            2.168266  2020  
...                 ...   ...  
1079           1.800000  2021  
1080           0.648000  2021  
1081           1.095000  2021  
1082           1.205000  2021  
1083           1.895000  2021  

[302 rows x 12 columns]

In [215]:
df[df.Year.isin([2016])]

Country                           Region  Happiness Rank  \
158      Denmark                   Western Europe             1.0   
159  Switzerland                   Western Europe             2.0   
160      Iceland                   Western Europe             3.0   
161       Norway                   Western Europe             4.0   
162      Finland                   Western Europe             5.0   
..           ...                              ...             ...   
310        Benin               Sub-Saharan Africa           153.0   
311  Afghanistan                    Southern Asia           154.0   
312         Togo               Sub-Saharan Africa           155.0   
313        Syria  Middle East and Northern Africa           156.0   
314      Burundi               Sub-Saharan Africa           157.0   

     Happiness Score  GDP per Capita  Social Support  Healthy Life Expectancy  \
158            7.526         1.44178         1.16374                  0.79504   
159            7.509         1.52733         1.14524                  0.86303   
160            7.501         1.42666         1.18326                  0.86733   
161            7.498         1.57744         1.12690                  0.79579   
162            7.413         1.40598         1.13464                  0.81091   
..               ...             ...             ...                      ...   
310            3.484         0.39499         0.10419                  0.21028   
311            3.360         0.38227         0.11037                  0.17344   
312            3.303         0.28123         0.00000                  0.24811   
313            3.069         0.74719         0.14866                  0.62994   
314            2.905         0.06831         0.23442                  0.15747   

     Freedom to Make Life Choices  Corruption Perception  Generosity  \
158                       0.57941                0.44453     0.36171   
159                       0.58557                0.41203     0.28083   
160                       0.56624                0.14975     0.47678   
161                       0.59609                0.35776     0.37895   
162                       0.57104                0.41004     0.25492   
..                            ...                    ...         ...   
310                       0.39747                0.06681     0.20180   
311                       0.16430                0.07112     0.31268   
312                       0.34678                0.11587     0.17517   
313                       0.06912                0.17233     0.48397   
314                       0.04320                0.09419     0.20290   

     Dystopia Residual  Year  
158            2.73939  2016  
159            2.69463  2016  
160            2.83137  2016  
161            2.66465  2016  
162            2.82596  2016  
..                 ...   ...  
310            2.10812  2016  
311            2.14558  2016  
312            2.13540  2016  
313            0.81789  2016  
314            2.10404  2016  

[157 rows x 12 columns]

In [216]:
c = df.loc[158:314].Country.tolist()
r = df.loc[158:314].Region.tolist()
region = {c[i] : r[i] for i in range(len(c))}
region
df['Region'] = df.Country.map(region)
df


Country                           Region  Happiness Rank  \
0     Switzerland                   Western Europe             1.0   
1         Iceland                   Western Europe             2.0   
2         Denmark                   Western Europe             3.0   
3          Norway                   Western Europe             4.0   
4          Canada                    North America             5.0   
...           ...                              ...             ...   
1225     Botswana               Sub-Saharan Africa           142.0   
1226       Rwanda               Sub-Saharan Africa           143.0   
1227     Zimbabwe               Sub-Saharan Africa           144.0   
1228      Lebanon  Middle East and Northern Africa           145.0   
1229  Afghanistan                    Southern Asia           146.0   

      Happiness Score  GDP per Capita  Social Support  \
0               7.587         1.39651         1.34951   
1               7.561         1.30232         1.40223   
2               7.527         1.32548         1.36058   
3               7.522         1.45900         1.33095   
4               7.427         1.32629         1.32261   
...               ...             ...             ...   
1225            3.471         1.50300         0.00100   
1226            3.268         0.00100         0.00000   
1227            2.995         0.00100         0.00100   
1228            2.955         1.39200         0.00000   
1229            2.404         0.00100         0.00000   

      Healthy Life Expectancy  Freedom to Make Life Choices  \
0                     0.94143                       0.66557   
1                     0.94784                       0.62877   
2                     0.87464                       0.64938   
3                     0.88521                       0.66973   
4                     0.90563                       0.63297   
...                       ...                           ...   
1225                  0.28000                       0.57100   
1226                  0.46200                       0.62100   
1227                  0.27000                       0.32900   
1228                  0.63100                       0.10300   
1229                  0.28900                       0.00000   

      Corruption Perception  Generosity  Dystopia Residual  Year  
0                   0.41978     0.29678            2.51738  2015  
1                   0.14145     0.43630            2.70201  2015  
2                   0.48357     0.34139            2.49204  2015  
3                   0.36503     0.34699            2.46531  2015  
4                   0.32957     0.45811            2.45176  2015  
...                     ...         ...                ...   ...  
1225                0.10200     0.01200            0.00000  2022  
1226                0.54400     0.18700            0.00100  2022  
1227                0.10500     0.10600            0.00100  2022  
1228                0.03400     0.08200            0.00000  2022  
1229                0.00500     0.08900            1.26300  2022  

[1230 rows x 12 columns]

In [217]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1230 entries, 0 to 1229
Data columns (total 12 columns):
 #   Column                        Non-Null Count  Dtype  
---  ------                        --------------  -----  
 0   Country                       1230 non-null   object 
 1   Region                        1180 non-null   object 
 2   Happiness Rank                1230 non-null   float64
 3   Happiness Score               1230 non-null   float64
 4   GDP per Capita                1230 non-null   float64
 5   Social Support                1230 non-null   float64
 6   Healthy Life Expectancy       1230 non-null   float64
 7   Freedom to Make Life Choices  1230 non-null   float64
 8   Corruption Perception         1229 non-null   float64
 9   Generosity                    1230 non-null   float64
 10  Dystopia Residual             918 non-null    float64
 11  Year                          1230 non-null   int32  
dtypes: float64(9), int32(1), object(2)
memory usage: 110.6+ KB


In [218]:
df.to_csv(r'C:/Users/Agata/Documents/GitHub/Happiness-Report/cleaned_data.csv', index=False)